compiler/src/main/java/com/squareup/anvil/compiler/codegen/BindingModuleGenerator.kt

```diff
@@ -237,6 +242,25 @@ internal class BindingModuleGenerator(
             checkExtendsBoundType(type = contributedClass, boundType = boundType)
             checkNotGeneric(type = contributedClass, boundTypeDescriptor = boundType)
 
+            Binding(contributedClass, annotation, boundType)
+          }
+          .let { bindings ->
+            if (isMultibinding) {
+              bindings
+            } else {
+              bindings
+                .groupBy {
+                  // It's safe to use the FqName here, because we don't allow generic super types.
+                  // If we wouldn't have this limitation, we'd need to use the Kotlin type.
+                  it.boundType.fqNameSafe
+                }
+                .map {
+                  it.value.findHighestPriorityBinding()
+                }
+                .asSequence()
```

Comment:
Will it fail faster (or use less memory) if we'll first convert to the `Sequence`, and then will do mapping?

Reply:
Everything in this longer sequence. But because of the `groupBy` above we get a Map and the `map` operator converts it to a sequence.